// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.sync.protocol";

option optimize_for = LITE_RUNTIME;

package sync_pb;

import "components/sync/protocol/password_sharing_invitation_specifics.proto";

// A message to obtain a list of recipients for sending a password.
message PasswordSharingRecipientsRequest {
  // The request returns family members for the user and that is why no
  // additional fields are required.
}

message PasswordSharingRecipientsResponse {
  enum PasswordSharingRecipientsResult {
    UNKNOWN = 0;

    // The user is a member of a family and the request succeeded.
    SUCCESS = 1;

    // Not a family member, used to distinguish from a family with
    // only one member.
    NOT_FAMILY_MEMBER = 2;
  }
  optional PasswordSharingRecipientsResult result = 1;

  // List of possible recipients for sending a password. Note that public key
  // may be absent if a recipient canâ€™t receive a password (e.g. due to an older
  // Chrome version).
  repeated UserInfo recipients = 2;
}
